src/test/java/org/javarosa/core/model/TriggerableDagTest.java

```diff
@@ -528,6 +528,40 @@ public void parsing_forms_with_cycles_should_fail() throws IOException {
         ));
     }
 
+    @Test
+    public void parsing_forms_with_cycles_by_self_reference_in_relevance_should_fail() throws IOException {
+        exceptionRule.expect(RuntimeException.class);
+        exceptionRule.expectMessage("Dependency cycles amongst the xpath expressions in relevant/calculate");
+
+        Scenario.init("Some form", html(head(
+            title("Some form"),
+            model(
+                mainInstance(t("data id=\"some-form\"",
+                    t("count", "1")
+                )),
+                bind("/data/count").type("int").relevant(". + 1")
```

Comment:
Relevants should be boolean expressions. It's interesting that `. + 1` is allowed and I'm not really sure what the behavior is. For clarity I think it'd be good to add e.g. `> 0`.